pycobertura/reporters.py

```diff
@@ -93,152 +67,166 @@ def __init__(self, *args, **kwargs):
+            self.differ.diff_line_rate(filename) for filename in self.differ_files_info
+        ]
 
+        indexes_of_files_with_changes = [
+            i
+            for i in range(len(self.differ_files_info))
+            if any(
+                (
+                    diff_total_stmts[i],
+                    diff_total_miss[i],
+                    diff_total_cover[i],
+                )
+            )
+        ]
 
-class TextReporterDelta(DeltaReporter):
-    def __init__(self, *args, **kwargs):
-        """
-        Takes the same arguments as `DeltaReporter` but also takes the keyword
-        argument `color` which can be set to True or False depending if the
-        generated report should be colored or not (default `color=False`).
-        """
-        self.color = kwargs.pop("color", False)
-        super(TextReporterDelta, self).__init__(*args, **kwargs)
+        filenames_of_files_with_changes = [
+            self.differ_files_info[i] for i in indexes_of_files_with_changes
+        ]
 
-    def format_row(self, row):
-        total_statements = "%+d" % row.total_statements if row.total_statements else "-"
-        line_rate = "%+.2f%%" % (row.line_rate * 100) if row.line_rate else "-"
-        total_misses = "%+d" % row.total_misses if row.total_misses else "-"
+        lines = {
+            "Filename": filenames_of_files_with_changes,
+            "Stmts": [
+                self.format_total_statements(diff_total_stmts[i])
+                for i in indexes_of_files_with_changes
+            ],
+            "Miss": [
+                self.format_total_misses(diff_total_miss[i])
+                for i in indexes_of_files_with_changes
+            ],
+            "Cover": [
+                self.format_line_rate(diff_total_cover[i])
+                for i in indexes_of_files_with_changes
+            ],
+        }
 
-        if self.color is True and total_misses != "-":
-            colorize = [green, red][total_misses[0] == "+"]
-            total_misses = colorize(total_misses)
+        lines["Filename"].extend(["TOTAL"])
+        lines["Stmts"].extend(
+            [self.format_total_statements(self.differ.diff_total_statements())]
+        )
+        lines["Miss"].extend(
+            [self.format_total_misses(self.differ.diff_total_misses())]
+        )
+        lines["Cover"].extend([self.format_line_rate(self.differ.diff_line_rate())])
 
-        if self.show_source is True:
-            missed_lines = [
-                "%s%d" % (["-", "+"][is_new], lno) for lno, is_new in row.missed_lines
+        if self.show_source:
+            diff_total_missing = [
+                self.differ.diff_missed_lines(filename)
+                for filename in self.differ_files_info
             ]
+            lines["Missing"] = [
+                self.format_missed_lines(diff_total_missing[i])
+                for i in indexes_of_files_with_changes
+            ]
+            lines["Missing"].extend([""])
```

Comment:
This would be a little cheaper since we don't have to construct a temporary array that would only hold the empty string:

```suggestion
            lines["Missing"].append("")
```

Reply:
done